packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -2081,79 +2079,48 @@ class _RenderAppBarTitleBox extends RenderAligningShiftedBox {
   }
 }
 
-enum _ScrollUnderFlexibleVariant { medium, large }
-
 class _ScrollUnderFlexibleSpace extends StatelessWidget {
   const _ScrollUnderFlexibleSpace({
     this.title,
     this.foregroundColor,
-    required this.variant,
-    this.primary = true,
+    required this.configBuilder,
     this.titleTextStyle,
     required this.bottomHeight,
   });
 
   final Widget? title;
   final Color? foregroundColor;
-  final _ScrollUnderFlexibleVariant variant;
-  final bool primary;
+  final _FlexibleConfigBuilder configBuilder;
   final TextStyle? titleTextStyle;
   final double bottomHeight;
 
   @override
   Widget build(BuildContext context) {
-    late final ThemeData theme = Theme.of(context);
     late final AppBarTheme appBarTheme = AppBarTheme.of(context);
-    final AppBarTheme defaults = theme.useMaterial3 ? _AppBarDefaultsM3(context) : _AppBarDefaultsM2(context);
+    late final AppBarTheme defaults = Theme.of(context).useMaterial3 ? _AppBarDefaultsM3(context) : _AppBarDefaultsM2(context);
```

Comment:
This is inside `build`. We might not need `late`.

```suggestion
   final AppBarTheme defaults = Theme.of(context).useMaterial3 ? _AppBarDefaultsM3(context) : _AppBarDefaultsM2(context);
```

Reply:
The `late` keywords are for lazy evaluation. When `foregroundColor` is specified, `defaults` won't be evaluated so this widget doesn't have to depend on the `Theme` widget.